front/pages/api/v1/w/[wId]/data_sources/[name]/tables/index.ts

```diff
@@ -169,6 +169,7 @@ async function handler(
       const tablesRes = await coreAPI.getTables({
         projectId: dataSource.dustAPIProjectId,
         dataSourceName: dataSource.name,
+        filter: getFilterFromQuery(req.query),
```

Comment:
Not sure I understand this? We have no guarantee to have a filter set on v1 endpoints?

We will need some data source views related endpoint instead no?

Reply:
I'm indeed not sure if it's useful to add the filters parameters in these public endpoints (still, they are completely optional, and were useful to test) - I've added them in core endpoints so just passing through here, but actually the filters might need to be generated from the vault to only display the tables the user has access to ?

Reply:
Let's remove them. We indeed don't want people to rely on them 🙏

Reply:
Eventually we will have data source view endpoints in the public API that apply filters (read operations) letting users with access to the view but not the data source reading from these views.

When interacting with dust apps we will pass the data source view id from which core will call the registry lookup to receive the filter and apply it.

Does it make sense. It's a pretty important part of the whole design?

We want to remove these because we always want to avoid unused code in public endpoint as users could start leveraging it and would prevent us from removing it without breaking their integrations

Reply:
For the filtering in blocks, all is done through the registry, so we don't need to do anything for these endpoints.
The list tables endpoints is used when creating a dust-app to select a table, for example. Should we filter the results based on the auth ?

Reply:
We'll have data source views endpoints for that right?

Reply:
+1 with having endpoints to list data sources/data source views that bundles the authentification logic.